### ISSUE TITLE
Sender name is not appear in the push notification for all activities

### ISSUE THREAD

**Issue by CometKim** [mattermost/platform#5416]
#### Summary
Sender name is not appear in the push notification for all activities, if the message does not contain mention.

#### Steps to reproduce
1. Set the push notification setting to receive for all activities.
2. Receive a push notification from messages with no mentions.

#### Expected behavior
`{sender} in {channel}: {message}` displayed on the push notification

#### Observed behavior
`in {channel}: {message}`

#### Possible fixes
sender name is built on this line, only for mentioned user list.
https://github.com/mattermost/platform/blob/master/app/notification.go#L112

~~https://github.com/mattermost/platform/blob/master/app/notification.go#L249~~
~~so, may this block is always skipped~~

https://github.com/mattermost/platform/blob/master/app/notification.go#L261
so, may the `senderName[id]` on this line is always empty

**Comment by lindy65** [comment]
Hi @CometKim 

Thanks for your report.  I tried reproducing the issue by setting "send mobile push notifications" to "all activity" but could not repro.  I also tried setting myself "away" and "offline" but still could not repro. I get the expected behaviour of "{sender} in {channel}: {message}".

Could you help by letting me know :
- Mattermost server version
- iOS device and iOS version
- was the message sent by a webhook or other integration?
- what is the name displayed in a desktop notification?
- does it reproduce for all users or only specific ones?

Once we have that information from you we can try reproducing the issue again.

Thanks!

**Comment by paav-o** [comment]
I can confirm this observed behavior noticed by CometKim and I'm happy to try to help debug it.

- We are running Mattermost v3.6.2 installed from Docker image and postgres as database. Build date is Tue Jan 31 21:01:44 UTC 2017.
- Test device is iPhone 6 with newest iOS 10.2.1 and running newest Mattermost v3.5 app from App Store (updated 18 Nov 2016).
- Message was sent by user manually typing it to the web app.
- Both the web app running on Firefox and the newest OS X desktop app (v3.5.0) show the lowercase 'Username' in their notifications (while sender has both 'Full Name' and 'Nickname' set in Account Settings).
- It does reproduce for all users.

Mobile Push settings in System Console:
```
Enable Push Notifications:  Use iOS and Android apps on iTunes and Google Play with TPNS
Push Notification Server:   http://push-test.mattermost.com
Push Notification Contents: Send full message snippet
```

**Comment by CometKim** [comment]
Thanks, @lindy65 and @paav-o.

Once you change your push notification setting to `for all activities`, It could be reproduced on pre-release like below.  

![screenshot_smartselect_2017-02-24-11-18-47](https://cloud.githubusercontent.com/assets/9696352/23287837/33eb3626-fa83-11e6-81c1-ae8c166dae61.png)

**Comment by CometKim** [comment]
Hello @lindy65, 

* Mattermost server version  
  recent version (pre-release)

* iOS device and iOS version  
  I use **Samsung Galaxy Note 5** and **Android Marshmallow**(6.0.1).
  I believe this is a server issue and not an app or device.

* was the message sent by a webhook or other integration?  
  General message and Integration both.

* what is the name displayed in a desktop notification?  
  I did not check the desktop notification, so I will report it later.

* does it reproduce for all users or only specific ones?
  For all users.

At least for me and my team, it's being reproduced by changing notification settings to `For all activities`.

Please let me know if the information is not enough to reproduce.

**Comment by lindy65** [comment]
Hi @CometKim and @paav-o 

Thank you for the detailed repro steps!  We've been able to repro the issue and have created a [ticket](https://mattermost.atlassian.net/browse/PLT-5685) to track it.

**Comment by jasonblais** [comment]
Thanks all! This should be fixed with Mattermost 3.7 on March 16th.
